File: clients/python-Component-Broker/examples/consume.py
#!/usr/bin/env python

from Broker.Messages import Message, Subscribe, Acknowledge
from Broker.Transport import TCP, UDP
from Broker.Codecs import Protobuf
from Broker.Clients import Minimal

server='localhost'
destination = '/python/tests'
destination_type = 'QUEUE'
N=10000

broker = Minimal(codec=Protobuf(), transport=TCP(host=server))

broker.send(Subscribe(destination=destination, destination_type=destination_type))
for n in xrange(N):
    message = broker.receive()
    broker.send(Acknowledge(message_id=message.message.id, destination=message.destination))
    print message.message.payload

File: clients/perl-component/examples/consume.pl
use lib ('../lib');

use SAPO::Broker::Clients::Simple;
use Data::Dumper;

use strict;
use warnings;

my $broker = SAPO::Broker::Clients::Simple->new(
    'host'  => 'localhost',
    'proto' => 'tcp'
);
my %options = (
    'destination_type' => 'TOPIC',
    'destination'      => '/tests/perl',
    'auto_acknowledge' => 1
);

$broker->subscribe(%options);

my $N = $ARGV[0] || 100;

for my $n ( 1 .. $N ) {
    my $notification = $broker->receive;
    my $payload      = $notification->message->payload;
    print $payload. "\n";
}
